### ISSUE TITLE
Embedding dimension error when using MultiClassifierDL in multi-label classification pipeline

### ISSUE THREAD

**Issue by fahminlb33** [JohnSnowLabs/spark-nlp#2948]
I'm trying to create a multi-label classification pipeline using `MultiClassifierDL` and BERT but after some training the process raised an exception of `IllegalArgumentException: requirement failed: The MultiClassifierDL only accepts embeddings larger than 1 and less than 1024 dimensions. Current dimension is 2048. Please use embeddings with at max 1024 dimensions`.

I'm running this script in Kaggle Notebook without GPU and using the default script `!wget http://setup.johnsnowlabs.com/kaggle.sh -O - | bash` to install Spark NLP.

Sample data:

![image](https://user-images.githubusercontent.com/8360880/119294150-032b4280-bc7e-11eb-8638-b493c7ade32f.png)

* Spark NLP version `sparknlp.version()`: 3.0.3
* Apache NLP version `spark.version`: 3.1.1
* Code to reproduce: https://gist.github.com/fahminlb33/d7186d89ceb0915d38ab456d099a4480

## Setup Kaggle Logs

```
--2021-06-02 00:24:40--  http://setup.johnsnowlabs.com/kaggle.sh
Resolving setup.johnsnowlabs.com (setup.johnsnowlabs.com)... 51.158.130.125
Connecting to setup.johnsnowlabs.com (setup.johnsnowlabs.com)|51.158.130.125|:80... connected.
HTTP request sent, awaiting response... 302 Moved Temporarily
Location: https://raw.githubusercontent.com/JohnSnowLabs/spark-nlp/master/scripts/kaggle_setup.sh [following]
--2021-06-02 00:24:40--  https://raw.githubusercontent.com/JohnSnowLabs/spark-nlp/master/scripts/kaggle_setup.sh
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1547 (1.5K) [text/plain]
Saving to: ‘STDOUT’

-                   100%[===================>]   1.51K  --.-KB/s    in 0s      

2021-06-02 00:24:40 (17.4 MB/s) - written to stdout [1547/1547]

setup Kaggle for PySpark 3.0.2 and Spark NLP 3.0.3
E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/universe/o/openjdk-8/openjdk-8-jre-headless_8u282-b08-0ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.88.142 80]
E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/universe/o/openjdk-8/openjdk-8-jdk-headless_8u282-b08-0ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.88.142 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  210M  100  210M    0     0  77.9M      0  0:00:02  0:00:02 --:--:-- 77.9M
```

## Logs

```
---------------------------------------------------------------------------
IllegalArgumentException                  Traceback (most recent call last)
<timed exec> in <module>

/opt/conda/lib/python3.7/site-packages/pyspark/ml/base.py in fit(self, dataset, params)
    127                 return self.copy(params)._fit(dataset)
    128             else:
--> 129                 return self._fit(dataset)
    130         else:
    131             raise ValueError("Params must be either a param map or a list/tuple of param maps, "

/opt/conda/lib/python3.7/site-packages/pyspark/ml/pipeline.py in _fit(self, dataset)
    107                     dataset = stage.transform(dataset)
    108                 else:  # must be an Estimator
--> 109                     model = stage.fit(dataset)
    110                     transformers.append(model)
    111                     if i < indexOfLastEstimator:

/opt/conda/lib/python3.7/site-packages/pyspark/ml/base.py in fit(self, dataset, params)
    127                 return self.copy(params)._fit(dataset)
    128             else:
--> 129                 return self._fit(dataset)
    130         else:
    131             raise ValueError("Params must be either a param map or a list/tuple of param maps, "

/opt/conda/lib/python3.7/site-packages/pyspark/ml/wrapper.py in _fit(self, dataset)
    319 
    320     def _fit(self, dataset):
--> 321         java_model = self._fit_java(dataset)
    322         model = self._create_model(java_model)
    323         return self._copyValues(model)

/opt/conda/lib/python3.7/site-packages/pyspark/ml/wrapper.py in _fit_java(self, dataset)
    316         """
    317         self._transfer_params_to_java()
--> 318         return self._java_obj.fit(dataset._jdf)
    319 
    320     def _fit(self, dataset):

/opt/conda/lib/python3.7/site-packages/py4j/java_gateway.py in __call__(self, *args)
   1303         answer = self.gateway_client.send_command(command)
   1304         return_value = get_return_value(
-> 1305             answer, self.gateway_client, self.target_id, self.name)
   1306 
   1307         for temp_arg in temp_args:

/opt/conda/lib/python3.7/site-packages/pyspark/sql/utils.py in deco(*a, **kw)
    132                 # Hide where the exception came from that shows a non-Pythonic
    133                 # JVM exception message.
--> 134                 raise_from(converted)
    135             else:
    136                 raise

/opt/conda/lib/python3.7/site-packages/pyspark/sql/utils.py in raise_from(e)

IllegalArgumentException: requirement failed: The MultiClassifierDL only accepts embeddings larger than 1 and less than 1024 dimensions. Current dimension is 2048. Please use embeddings with at max 1024 dimensions
```

**Comment by maziyarpanahi** [comment]
Hi,

Could you please make sure you have a unified input flow for all the annotators and try again? 

You have to either go with `document` or `sentence`, but not mixed. Not sure if this is the case, something else can be going on but I would like to test it with your dataset to have `sentence` for everything and also removing `setDimension` since those play no role and already set:


```python
document = DocumentAssembler()\
  .setInputCol("text")\
  .setOutputCol("document")

sentence_detector = SentenceDetectorDLModel\
  .pretrained("sentence_detector_dl", "en") \
  .setInputCols(["document"]) \
  .setOutputCol("sentences")

tokenizer = Tokenizer() \
    .setInputCols(["sentences"]) \
    .setOutputCol("token")

embeddings = BertEmbeddings.pretrained("covidbert_large_uncased", "en") \
        .setInputCols("sentences", "token") \
        .setOutputCol("embeddings")

sentence_embeddings = SentenceEmbeddings() \
        .setInputCols(["sentences", "embeddings"]) \
        .setOutputCol("sentence_embeddings") \
        .setPoolingStrategy("AVERAGE")

```

**Comment by maziyarpanahi** [comment]
There was an issue with `SentenceEmbeddings` so it should be fixed after Spark NLP 3.3.0 release
